mobius/src/geometry/line.rs

```diff
@@ -11,14 +50,40 @@ pub struct Line {
 impl Line {
     /// Create a line from a normal and distance offset
     /// This will automatically normalize the normal
-    pub fn new(normal: Complex, distance: f64) -> Result<Self, String> {
-        match normal.normalize() {
-            Some(unit_normal) => Ok(Self {
-                unit_normal,
-                distance,
-            }),
-            None => Err(String::from("Normal must be finite and non-zero")),
+    pub fn new(normal: Complex, distance: f64) -> Result<Self, LineError> {
```

Comment:
I don't like the way this constructor works, because it only normalizes the normal, not the distance value as well. rethink this..

Reply:
Thinking about it, if you divide the normal by its magnitude, the distance has to be divided by the same amount. Add a unit test and implementation for this.